src/GraphQL/Validation/DocumentValidator.ParseArgumentVisitor.cs

```diff
@@ -0,0 +1,183 @@
+            var fragmentName = fragmentSpread.FragmentName.Name.Value;
+            if ((context.VisitedFragments ??= new()).Add(fragmentName))
+            {
+                foreach (var definition in context.ValidationContext.Document.Definitions)
+                {
+                    // the fragment name should be one of the definitions in the document, or the
+                    // document would not have passed validation
+                    if (definition is GraphQLFragmentDefinition fragment && fragment.FragmentName.Name.Value == fragmentName)
+                    {
+                        return VisitFragmentDefinitionAsync(fragment, context);
+                    }
+                }
+            }
+            return default;
+        }
+
+        protected override async ValueTask VisitFragmentDefinitionAsync(GraphQLFragmentDefinition fragmentDefinition, Context context)
+        {
+            // the fragment type name should be listed in the document, or the document would have failed validation
+            if (context.Schema.AllTypes[fragmentDefinition.TypeCondition.Type.Name.Value]?.GetNamedType() is IComplexGraphType type)
+            {
+                context.Types.Push(type);
+                await base.VisitFragmentDefinitionAsync(fragmentDefinition, context).ConfigureAwait(false);
+                context.Types.Pop();
+            }
+        }
+
+        protected override async ValueTask VisitFieldAsync(GraphQLField field, Context context)
+        {
+            // find the field definition for this field
+            var schema = context.Schema;
+            var fieldType =
+                field.Name.Value == schema.TypeMetaFieldType.Name ? schema.TypeMetaFieldType :
+                field.Name.Value == schema.TypeNameMetaFieldType.Name ? schema.TypeNameMetaFieldType :
+                field.Name.Value == schema.SchemaMetaFieldType.Name ? schema.SchemaMetaFieldType :
+                context.Type?.Fields.Find(field.Name.Value);
+            // should not be null, or the document would have failed validation
+            if (fieldType == null)
+                return;
+            // if any arguments were supplied in the document for the field, load all defined arguments
+            // for the field
+            if (field.Arguments?.Count > 0)
+            {
+                try
+                {
+                    var arguments = ExecutionHelper.GetArguments(fieldType.Arguments, field.Arguments, context.Variables);
+                    if (arguments != null)
+                    {
+                        (context.ArgumentValues ??= new()).Add(field, arguments);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    // todo: report error properly
+                    context.ValidationContext.ReportError(new ValidationError($"Error trying to resolve field '{field.Name.Value}'.", ex));
```

Comment:
Note to self: Update this with a more appropriate error message.  Currently set this way because that was the error message that would have occurred previously.

Reply:
another pr?